### DISCUSSION TITLE
Help setting up pr-previews

### DISCUSSION THREAD

**Discussion by IamShubhamGupto** [rossjrw/pr-preview-action#48] [Q&A] [answered]
Hey

I am trying to use pr-previews for one of my projects - https://github.com/compressed-sensing/algorithms/pull/11

I want my production website `gh-pages` branch and preview website `preview-{pr-num}` branch to be separate as I don't want the original production website to be overwritten whenever I create a pr

I also want multiple prs to have previews available along with the production website running so that we can compare. 

Could you help me out? I think its doable with your action but Im struggling to set it up

**Comment by rossjrw** [comment]
The answer to this depends on what you mean by 'website'.

The default use case is that your main deployment lives in the root directory on the `gh-pages` branch, and your preview deployments live in `pr-preview/pr-XX/` on the `gh-pages` branch. When you go to https://compressed-sensing.github.io/algorithms, you get the homepage of your main deployment, and when you go to https://compressed-sensing.github.io/algorithms/pr-preview/pr-XX/ you get the homepage of that preview. They both exist at the same time, as totally different webpages, but ostensibly on the same website (compressed-sensing.github.io).

If you want an entire other domain for your preview (e.g. compressed-sensing-algorithms-pr-preview-XX.github.io), GitHub Pages doesn't really support that - you can only have one GitHub Pages site per organisation, so you'll need to make a new organisation account on GitHub just for that website. That doesn't make much sense to me.

If you want a completely isolated environment for your previews, it might make sense to use a totally seperate service for those deployments, like Vercel or Netlify.

I think you just want to make sure that you can use this Action to have multiple PR previews _and_ your main deployment on the same branch at the same time. If so, the answer to that question is yes you can; that's the whole point of this action.

**Reply by IamShubhamGupto** [reply]
@rossjrw 
"I think you just want to make sure that you can use this Action to have multiple PR previews and your main deployment on the same branch at the same time. If so, the answer to that question is yes you can; that's the whole point of this action."

yes, this is exactly what I want. I changed my `deploy.yml` and `preview.yml` yesterday based on the guidelines in `README.md` of your action. Could you verify if what ive written is correct?

Thank you

**Reply by rossjrw** [reply, marked as answer]
@IamShubhamGupto Yep, looks approximately right to me:

https://github.com/compressed-sensing/algorithms/blob/cbf887bd7d0c0bad45ae3e603098dd405b764e63/.github/workflows/preview.yml#L62C2-L65

```yml
    - name: 🚀 Deploy Preview to GitHub Pages
      uses: rossjrw/pr-preview-action@v1
      with:
        source-dir: _build/html
```

Best way to test it will be to create a PR now and see if the preview works as expected.

I notice in your `gh-pages` branch there's a stale preview for a closed PR that wasn't cleaned up: https://github.com/compressed-sensing/algorithms/tree/gh-pages/pr-preview/pr-11 - I suspect this is from when you were trialling different preview methods and turned off my action before it had a chance to clean up after itself. If that bothers you, you might want to nip in and delete it. It's currently accessible at https://compressed-sensing.github.io/algorithms/pr-preview/pr-11/intro.html

**Comment by IamShubhamGupto** [comment]
Hey @rossjrw 

I really liked how the action worked and I wanted to apply it to other projects of mine.

Here is a PR for a repository that I have raised from a forked copy

https://github.com/m2lines/L96_demo/actions/runs/4894707073/jobs/8739302543?pr=111

as you can see in the logs, it fails due to `Error: Resource not accessible by integration`

we found that if the same code is written and raised from the same repository but different branch, then the action works as expected. Only when there's a PR coming from a fork, it fails. 

Any way to go around this?

Thanks

**Reply by rossjrw** [reply]
@IamShubhamGupto That's correct, PR Preview Action does not currently support previews from forks. See https://github.com/rossjrw/pr-preview-action/pull/6 for details.

There _are_ ways around it - chiefly, you can use the pull_request_target event instead of pull_request, but this opens you up to huge security concerns - now anyone can fork your repository and deploy code to it, without you knowing anything about it. I don't recommend using this action for that use case. I recommend using a dedicated service instead, such as Vercel or Netlify.

**Reply by IamShubhamGupto** [reply]
Hey @rossjrw 
Thanks for letting us know, we'll look at alternatives!
